### DISCUSSION TITLE
Is it possible to disable/enable a menu item in the contributes.commands.menus.editor.title?

### DISCUSSION THREAD

**Discussion by reregaga** [microsoft/vscode-discussions#375] [Extension Development QnA] [answered]
I'm making an extension and using the display [menu](https://code.visualstudio.com/api/references/contribution-points#contributes.menus) `"editor/title"` in editor's title bar and would like to be able to hide this:  

>![image](https://user-images.githubusercontent.com/76629445/210848754-350d32da-30c6-4b4a-a7e2-02d81861f23b.png)

element in some cases.  
So far I've only found a way to hide it by changing `package.json`, but I don't think that's the right way.  
Is there some API for this?

**Comment by alefragnani** [comment, marked as answer]
Hi @reregaga ,

You could use [When clause contexts](https://code.visualstudio.com/api/references/when-clause-contexts), as it allows you to customize _when_ such menu should be visible or not?

If you need enable/disable instead, look for `enablement` conditions (https://code.visualstudio.com/api/extension-guides/command#enablement-of-commands)

Hope this helps

**Reply by reregaga** [reply]
Thanks for the link!  
I found  [setting in when clause](https://code.visualstudio.com/api/references/when-clause-contexts#check-a-setting-in-a-when-clause) or [custom when clause](https://code.visualstudio.com/api/references/when-clause-contexts#add-a-custom-when-clause-context)  to be the right way to do it.
